tensorflow_graphics/datasets/shapenet/shapenet.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Lint as: python3
+"""Shapenet Core dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import csv
+import json
+import os
+import textwrap
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets as tfds
+from tensorflow_datasets import features as tfds_features
+from tensorflow_graphics.datasets import features as tfg_features
+
+_CITATION = """
+@techreport{shapenet2015,
+  title       = {{ShapeNet: An Information-Rich 3D Model Repository}},
+  author      = {Chang, Angel X. and Funkhouser, Thomas and Guibas, Leonidas and Hanrahan, Pat and Huang, Qixing and Li, Zimo and Savarese, Silvio and Savva, Manolis and Song, Shuran and Su, Hao and Xiao, Jianxiong and Yi, Li and Yu, Fisher},
+  number      = {arXiv:1512.03012 [cs.GR]},
+  institution = {Stanford University --- Princeton University --- Toyota Technological Institute at Chicago},
+  year        = {2015}
+}
+"""
+
+_DESCRIPTION = """
+ShapeNetCore is a densely annotated subset of ShapeNet covering 55 common object
+categories with ~51,300 unique 3D models. Each model in ShapeNetCore is linked
+to an appropriate synset in WordNet (version 3.0).
+
+The synsets will be extracted from the taxonomy.json file in the ShapeNetCore.v2.zip
+archive and the splits from http://shapenet.cs.stanford.edu/shapenet/obj-zip/SHREC16/all.csv
+"""
+
+_TAXONOMY_FILE_NAME = 'taxonomy.json'
+
+_MODEL_SUBPATH = os.path.join('models', 'model_normalized.obj')
+
+_SPLIT_FILE_URL = \
+    'http://shapenet.cs.stanford.edu/shapenet/obj-zip/SHREC16/all.csv'
+
+_CHECKSUMS_DIR = os.path.normpath(
+    os.path.join(os.path.dirname(__file__), 'checksums/'))
+
+
+class Shapenet(tfds.core.GeneratorBasedBuilder):
+  """ShapeNetCore V2.
```

Comment:
In the interest of making this extendable using the same namespace (e.g. adding a point-cloud variant) can this be implemented with a default `BUILDER_CONFIGS = [MeshConfig()]` ? MeshConfig itself can be a trivial `BuilderConfig` and can change later (other than changing the name), but adding `BUILDER_CONFIGS` later changes the directories in which things are saved, so would be a breaking change.